cpp/src/arrow/array-test.cc

```diff
@@ -2562,11 +2561,11 @@ TEST_P(DecimalTest, WithNulls) {
   std::vector<Int128> draw = {Int128(1),  Int128(2), Int128(-1), Int128(4),
                               Int128(-1), Int128(1), Int128(2)};
   Int128 big;
-  ASSERT_OK(DecimalUtil::FromString("230342903942.234234", &big));
+  ASSERT_OK(Int128::FromString("230342903942.234234", &big));
   draw.push_back(big);
 
   Int128 big_negative;
-  ASSERT_OK(DecimalUtil::FromString("-23049302932.235234", &big_negative));
+  ASSERT_OK(Int128::FromString("-23049302932.235234", &big_negative));
```

Comment:
For readability I think we need to change this type to be `Decimal128`. Maybe change the current int128.h/cc to be decimal.h/cc, since int128 in this library is really only an implementation detail of decimals